helpers/test/plugins.test.js

```diff
@@ -0,0 +1,67 @@
+import { getPlugin, getPlugins, registerPlugin } from '../plugins.js';
+import { expect } from '@brightspace-ui/testing';
+
+describe('plugins', () => {
+
+	before(() => {
+		registerPlugin('test-plugins', { prop1: 'beer' });
+		registerPlugin('test-plugins', { prop1: 'donuts' });
+		registerPlugin('test-plugins-sorted', { prop1: 'beer' }, { sort: 2 });
+		registerPlugin('test-plugins-sorted', { prop1: 'donuts' }, { sort: 1 });
+		registerPlugin('test-plugins-keyed', { prop1: 'beer' }, { key: 'plugin1' });
+		registerPlugin('test-plugins-keyed', { prop1: 'donuts' }, { key: 'plugin2' });
+	});
+
+	it('getPlugins should return empty array for invalid plugin set key', () => {
+		const plugins = getPlugins('invalid-plugin-set-key');
+		expect(plugins.length).to.equal(0);
+	});
+
+	it('getPlugins should return array of plugins in registration order', () => {
+		const plugins = getPlugins('test-plugins');
+		expect(plugins.length).to.equal(2);
+		expect(plugins[0].prop1).to.equal('beer');
+		expect(plugins[1].prop1).to.equal('donuts');
+	});
+
+	it('getPlugins should return array of plugins in sort order', () => {
+		const plugins = getPlugins('test-plugins-sorted');
+		expect(plugins.length).to.equal(2);
+		expect(plugins[0].prop1).to.equal('donuts');
+		expect(plugins[1].prop1).to.equal('beer');
+	});
+
+	it('getPlugins should return array of plugins in new sort order if additional plugins are registered', () => {
+		registerPlugin('test-plugins-sorted-add', { prop1: 'beer' }, { sort: 3 });
+		registerPlugin('test-plugins-sorted-add', { prop1: 'donuts' }, { sort: 1 });
+
+		let plugins = getPlugins('test-plugins-sorted-add');
+		expect(plugins.length).to.equal(2);
+		expect(plugins[0].prop1).to.equal('donuts');
+		expect(plugins[1].prop1).to.equal('beer');
+
+		registerPlugin('test-plugins-sorted-add', { prop1: 'candy apple' }, { sort: 2 });
```

Comment:
Question: wondering if registering a plugin _after_ someone has asked for the plugins should be an error state and `throw`? The way we want this working is for BSI/whatever to call into all the registrations first thing before any other code runs, so if someone registers later that to me implies they're doing something naughty.

Reply:
Haha, you just reminded me of Austin Powers. If we're gonna throw an error for being naughty, we should at least give them an error message in "English English". 😆 

Anyway, good question. The only scenario I can think of is if someone tried to register new plugin as part of dynamically loading a bunch more stuff after the initial page load. But given that consumers would not be notified that their plugin array is stale, it seems like a muddled situation to say the least.

Reply:
Updated to throw error if `registerPlugin` is called after `getPlugins` for the same plugin Set key.